lib/new_relic/agent.rb

```diff
@@ -649,6 +649,43 @@ def add_custom_span_attributes(params)
       end
     end
 
+    # Add custom attributes to log events for the current agent instance.
+    #
+    # @param [Hash] params    A Hash of attributes to attach to log
+    #                         events. The agent accepts up to 240 custom
+    #                         log event attributes.
+    #
+    #                         Keys will be coerced into Strings and must
+    #                         be less than 256 characters. Keys longer
+    #                         than 255 characters will be truncated.
+    #
+    #                         Values may be Strings, Symbols, numeric
+    #                         values or Booleans and must be less than
+    #                         4095 characters. If the value is a String
+    #                         or a Symbol, values longer than 4094
+    #                         characters will be truncated. If the value
+    #                         exceeds 4094 characters and is of a
+    #                         different class, the attribute pair will
+    #                         be dropped.
+    #
+    #                         This API can be called multiple times.
+    #                         If the same key is passed more than once,
+    #                         the value associated with the last call
+    #                         will be preserved.
```

Comment:
These comments seem to suggest that calling the API multiple times will have an additive effect (hash merge). Is there any need to a) reveal what the current set of attributes is and/or b) offer the ability to remove existing attributes?

Reply:
I'm not sure where I'd reveal the current set of attributes to a user in a meaningful way. Our logs are the only way we communicate with our customers. I could log a message with the current attributes each time a change is made. Would you like me to add that?

To remove existing attributes, a user could submit a `nil` or empty value to an existing key:
`NewRelic::Agent.add_custom_log_attributes(existing: nil)` or `NewRelic::Agent.add_custom_log_attributes(existing: '')`

Would you like me to describe how to remove an attribute with the current functionality? Is that suggestion too janky and it needs an API of its own? 

I'm also a bit confused about the case where you'd want to remove an attribute. You call the API to add the attribute, wouldn't you remove the API call or the key/value from the API call if you didn't want the attribute to be sent anymore?

Reply:
It seems like removing attributes doesn't seem like something we should put much effort into support. No one is requesting that currently, so if someone does want it they can always submit a feature request for it, but otherwise it seems like unnecessary effort to add that functionality to the existing PR.

Reply:
I agree with Tanna and would also prefer not to add an attribute removal API at this time.

@fallwith, would you like me to change the return value here to be the current attributes?